src/KubernetesClient/Cache/ThreadSafeStore.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.Threading;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace k8s.cache
+{
+    public class ThreadSafeStore : IThreadSafeStore
+    {
+        private static Mutex _storageLock;
```

Comment:
Why not just use `ConcurrentDictionary`?

Reply:
Originally, wanted to do multiple operations under the lock hence choose the mutex approach. But looking at now, should be possible to use ConcurrentDictionary. Will do an iteration with ConcurrentDictionary.